### ISSUE TITLE
OpenSSLInitializer::uninitialize() not calling OSSL_PROVIDER_unload for OpenSSL v3 leaving memory leak

### ISSUE THREAD

**Issue by yezenyounan** [pocoproject/poco#3945]
I'm using `Poco v1.12.4` and `OpenSSL v3.0.7` through `vcpkg v2023.01.09`, building on Ubuntu.
In my application I call `Poco::Net::initializeSSL()` in a class constructor and `Poco::Net::uninitializeSSL()` in the class's destructor.
`Poco::Net::initializeSSL()` and `Poco::Net::uninitializeSSL()` call `OpenSSLInitializer::initialize()` and `OpenSSLInitializer::uninitialize()`, respectively.
The `initialize()` function calls `OSSL_PROVIDER_load` for OpenSSL v3, but the `uninitialize()` function does NOT call `OSSL_PROVIDER_unload` for OpenSSL v3.
This causes my application to have memory leaks.
Is there a work around?

**Comment by obiltschnig** [comment]
No, we tried calling `OSSL_PROVIDER_unload()`, but this leads to a crash.
